### PR TITLE
Test evidence support for Ledger API tests [DPP-965]

### PR DESCRIPTION
**Pull request by skisel-da**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by mziolekda — COMMENTED** [review]
Left two comments

File: ledger/test-evidence/BUILD.bazel [hunk 1]
```diff
@@ -18,12 +16,33 @@ da_scala_binary(
         "@maven//:io_circe_circe_core",
         "@maven//:io_circe_circe_generic",
     ],
+    tags = ["maven_coordinates=com.daml:ledger-test-evidence-generator-lib:__VERSION__"],
```
[the remark this answers is not in the document]
Comment (skisel-da) [reply ledger/test-evidence/BUILD.bazel]: You're right, not supposed to be used externally, removing

File: ledger/test-evidence/src/main/scala/com/daml/test/evidence/generator/Main.scala [hunk 2]
```diff
@@ -33,29 +35,36 @@ object Main {
...
       .testEntries[TT, TS, TE](scalaTestSuites, testEntry)
+      .concat(LedgerApiTestGeneratorSupport.testEntries(ledgerApiSuites, testEntry))
 
   def main(args: Array[String]): Unit = {
     val runpathList: List[String] = loadClasspath()
       .map(_.split(":").toList)
       .getOrElse(List.empty)
 
+    val ledgerApiTests: List[LedgerTestSuite] =
+      (suites.v1_8.default(0L) ++ suites.v1_8.optional()).toList
```
[the remark this answers is not in the document]
Comment (skisel-da) [reply ledger/test-evidence/src/main/scala/com/daml/test/evidence/generator/Main.scala]: Added v1_14 as well, but it looks like currently v1_14 includes tests of v1_8, so how we would have same test exported twice? Shall we filter unique test suites?

[the remark this answers is not in the document]
Comment (skisel-da) [reply ledger/test-evidence/src/main/scala/com/daml/test/evidence/generator/Main.scala]: Changed to v1_14 as agreed over the call

**Review by mziolekda — APPROVED** [review]

**Pull request merged**